app/repositories/slick/implementations/SlickChatsRepository.scala

```diff
@@ -259,24 +286,289 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
 
-  private[implementations] def insertEmail(createEmailDTO: CreateEmailDTO, chatId: String,
+  /**
+   * Method that returns an action containing an instance of the class Email
+   * @param userId ID of the user
+   * @param emailId ID of the email
+   * @return a DBIOAction containing an instance of the class Email
+   */
+  private def getEmailAction(userId: String, emailId: String) = {
+    for {
+      email <- EmailsTable.all
+        .join(UserChatsTable.all)
+        .on {
+          case (emailRow, userChatRow) => emailRow.emailId === emailId && userChatRow.userId === userId &&
+            userChatRow.chatId === emailRow.chatId
+        }
+        .map { case (emailRow, _) => (emailRow.emailId, emailRow.body, emailRow.date, emailRow.sent) }
+        .result
+
+      attachmentIds <- AttachmentsTable.all.filter(_.emailId === emailId).map(_.attachmentId).result
+
+      emailAddresses <- EmailAddressesTable.all
+        .join(AddressesTable.all)
+        .on {
+          case (emailAddress, address) => emailAddress.emailId === emailId &&
+            emailAddress.addressId === address.addressId
+        }
+        .map { case (emailAddress, address) => (emailAddress.emailId, emailAddress.participantType, address.address) }
+        .result
+
+      groupedEmailAddresses = emailAddresses
+        .groupBy { case (thisEmailId, participantType, _) => (thisEmailId, participantType) }
+        .mapValues(_.map(_._3))
+
+    } yield buildEmailDto(email, groupedEmailAddresses, Map(emailId -> attachmentIds))
+  }
+
+  /**
+   * Method that, given a sequence of userIds and a chat, returns a Map with the UserChatRows of each user
+   * for that chat (if it exists) with their respective userIds as key
+   * @param userIds sequence of userIds
+   * @param chatId ID of the chat
+   * @return a DBIOAction that returns a Map with the userIds as key and UserChatRows of each user
+   * for that chat as value
+   */
+  private def getUserChatsByUserId(userIds: Seq[String], chatId: String): DBIOAction[Map[String, UserChatRow], NoStream, Effect.Read] = {
+    for {
+      userIdUserChatTuple <- UserChatsTable.all
+        .filter(userChat => userChat.userId.inSet(userIds) && userChat.chatId === chatId)
+        .map(userChat => userChat.userId -> userChat)
+        .result
+    } yield userIdUserChatTuple.toMap
+  }
+
+  /**
+   * Method that sends an email by:
+   * - inserting or updating the userChat of the receivers (users) to "inbox"
+   * - updating the email status to "sent"
+   * - updating the userChat of the sender (user) to "sent"
+   * @param senderUserId userId of the sender
+   * @param chatId ID of the chat
+   * @param emailId ID of the email
+   * @param addresses addresses of the receivers of the email
+   * @return a count of all the updated rows
+   */
+  private def sendEmail(senderUserId: String, chatId: String, emailId: String, addresses: Set[String]) = {
+    if (addresses.nonEmpty) {
```

Comment:
Please make it clear that this method returns an Action, either by changing it's name, specifying it's return type, or pointing it out in it's description.

Note: If you are avoiding writing the DBIOAction[R,S,E] return types because they are too long, you can write instead the shorter DBIO[R].

Reply:
I'll do that! Thanks :)